client/dist/page.js

```diff
@@ -22,34 +22,51 @@ function finishedWithNewGame() {
         });
         document.querySelector(selector).appendChild(gameList);
     });
-}
+}*/
 
 /**
  * This function handles the new game button event
  *
  **/
-function handleNewGame(selector) {
-    p = jg.newGame('#meet');
-    p.then(() => {
-        console.log(' a new game!!!');
-        finishedWithNewGame(jg);
-    })
-    .catch(() => {
-        alert('it failed!');
-    })
-    .finally(() => {
-        console.log('runs no matter what');
-    });
-}
 document.addEventListener('DOMContentLoaded', () => {
+
+
     console.log('document is ready. I can start now');
-    jg = new JitsiGame(config);
-    ttt = new TicTacToe()
-    jg.gameRoomLobby('#meet');
+    let jg = new JitsiGame(config);
+    const playerSession = jg.handlePlayerSession();
+    const ttt = new TicTacToe(jg, playerSession);
+
+    jg = new JitsiGame(config, ttt);
```

Comment:
Looks like jg gets assigned twice here, once above and then once here.  Was that intentional?

Reply:
It was. I wanted to use the renderGame method from the TicTacToe class inside of the JitsiGame class so I decided to pass it through as an argument. I couldn't do this when I first declared the variable jg because ttt wasn't stored in memory yet so I couldn't pass it as an argument. But I still needed a JitsiGame class to pass as an argument to the ttt constructor. This got it working but I would like to do something different here if you have any suggestions.

Reply:
Yeah I'm thinking probably we need to break out the data class into an external class and pass that in.  Since all we use is save game state and load game state, it might fit somewhere else than JitsiGame.  I'll have to think about it a bit more